src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -36,7 +36,7 @@ public void Add(Feature feature)
             }));
         }
 
-        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings)
+        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings, RoutingComponent routing)
```

Comment:
@SzymonPobiega 
1. We have a specific use case which we can see this approach bringing value and impact instead of being a refactoring.
2. That would increase the size of this PR by a large amount. The idea was to try this a few times and then see, whether we're still happy with the idea and whether we may be able to generalize some of the repeating logic. Small steps :)
3. not sure what the problem would be with 10 of these components?
4. absolutely, but we can easily do that in a dedicated PR 👍